test/common/http/conn_manager_impl_fuzz_test.cc

```diff
@@ -423,17 +424,19 @@ class FuzzStream {
         fakeOnData();
         FUZZ_ASSERT(testing::Mock::VerifyAndClearExpectations(config_.codec_));
         state = StreamState::Closed;
+        decoding_done_ = false;
```

Comment:
Why does the reset of `decoding_done_` not needed for the `kReturnDecoderError` case?
Is it because the codec implementation does some internal state reset?

Reply:
I'm generally wondering what we get by resetting this state variable on errors and other terminal conditions (including trailers), but I guess it doesn't hurt and might make things more understandable?

Reply:
What would make sense most in the `kReturnDecoderError`? Have `decoding_done_ = true;`, because on an error no recovery is possible, because the stream is closed anyway? Or should it rather be `= false;`, but why?

Reply:
Additional commit proposes `decoding_done_ = true;` in error case. Fuzzer running for an hour was happy.

Reply:
I don't mind either way, just thinking from a fuzzer's perspective, it may lead to some false crashes/issues if the state isn't cleared on error.